demos/test-app/lib.rs

```diff
@@ -216,7 +216,7 @@ lazy_static! {
 
         // the string we are replacing here is inserted by webpack during the front-end build
         let index_html = index_html.replace(
-            r#"<script defer="defer" src="bundle.js"></script>"#,
```

Comment:
I'm a little confused, how was the js injected?

Reply:
I'm confused now, too. Isn't the comment above this line exactly answering the question?
`// the string we are replacing here is inserted by webpack during the front-end build`

Reply:
The diff is there, because apparently webpack changed behavior and is now injecting something different (i.e. just `defer` instead of `defer="defer"`).

Reply:
> Isn't the comment above this line exactly answering the question?

It is... kind of 🙃 but I was wondering _how_ it was being injected if the string didn't match. Don't we use the test app in CI? If so, how does it work?

Reply:
Ok, I took another look. It works for the selenium tests because the value is not used for anything critical. This being broken just results in a silly default value of `[object HTMLInputElement]` in the canisterId input box.

But our test always fill in the canisterId explicitly so they are not affected by the default.